src/RqlParserException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * Exception handling
+ */
+namespace Graviton\Rql;
+
+/**
+ * @author  List of contributors <https://github.com/libgraviton/php-rql-parser/graphs/contributors>
+ * @license  http://opensource.org/licenses/gpl-license.php GNU Public License
+ * @link     http://swisscom.ch
+ */
+class RqlParserException extends \Exception
+{
+    const VISITOR_INTERFACE_NOT_IMPLEMENTED = 412;
+    const VISITOR_NOT_SUPPORTED = 405;
+}
```

Comment:
It looks as if this class is only being used in the bundle even though the ``VISITOR_INTERFACE_NOT_IMPLEMENTED`` case should be taken care of in the lib here.

We need to keep the ``VISITOR_NOT_SUPPORTED`` case in the bundle due to the fact that different users will have different di-containers that can do this for them.

Maybe it would even make sense to wrap the basic visitor into something that integrates with doctrine at a another level. That would mean moving the visitor to his own repo (as to not get to much deps on this one) and hooking into the doctrine filter/event thing.

I'm not sure where we should got with all this, but I'm pretty sure that it will make sense to move the mongodb parts to their own repo (as mentioned in the description of #11).

Some of these points aren't things that make sense taking care of now. I'm jotting them down for your consideration and to get us evolving this in the same general direction.